### PR TITLE
Angr management UI fixes

### PR DESCRIPTION
**Pull request by frqmod**
Changes:

- Added status bar to replace old connection status label (which has been removed)
- Updated config dialog to no longer allow "__root__" as a username
- Added indicator to status bar to show what the current context is (symbol@addr)
- Fixed the file dialog being finnicky when running as a SNAP package (e.g. in pycharm)

### CODE REVIEW

File: plugins/angr_binsync/controller.py [hunk 1]
```diff
@@ -35,10 +35,10 @@ def active_context(self):
         except NotImplementedError:
             return None
 
-        if func is None:
+        if func is None or func.am_obj is None:
             return None
 
-        return func.addr
+        return func.name, func.addr
```
Comment (mahaloz) [inline plugins/angr_binsync/controller.py:41]: @frqmod I don't like this checking for `func.name`. Can we instead make this only the func.addr again?

File: binsync/common/controller.py [hunk 2]
```diff
@@ -173,10 +174,10 @@ def start_pull_routine(self):
         self.pull_thread.start()
 
     def _check_and_notify_ctx(self):
-        active_ctx = self.active_context()
+        active_ctx_name, active_ctx = self.active_context()
```
Comment (mahaloz) [inline binsync/common/controller.py:177]: see my other comment about this, as I don't see why we need both name and ctx. We should only need ctx.

**Review by mahaloz — CHANGES_REQUESTED** [review]

File: plugins/angr_binsync/controller.py (hunk 1 above)
Comment (frqmod) [reply plugins/angr_binsync/controller.py]: yes

File: binsync/common/controller.py (hunk 2 above)
Comment (frqmod) [reply binsync/common/controller.py]: agreed

File: binsync/common/controller.py [hunk 3]
```diff
@@ -104,6 +104,7 @@ def __init__(self, headless=False):
         self.ctx_change_callback = None  # func()
         self._last_reload = datetime.datetime.now()
         self.last_ctx = None
+        self.last_ctx_name = None
```
Comment (mahaloz) [inline binsync/common/controller.py:107]: remove this

File: binsync/common/controller.py [hunk 4]
```diff
@@ -176,7 +177,6 @@ def _check_and_notify_ctx(self):
         active_ctx = self.active_context()
         if active_ctx is None or self.last_ctx == active_ctx:
             return
-
```
Comment (mahaloz) [inline binsync/common/controller.py:179]: put this space back

**Review by mahaloz — CHANGES_REQUESTED** [review]

File: binsync/common/controller.py (hunk 3 above)
Comment (frqmod) [reply binsync/common/controller.py:107]: oops

**Review by mahaloz — APPROVED** [review]

**Pull request merged**